services/catalog/src/simcore_service_catalog/api/routes/services.py

```diff
@@ -155,7 +155,7 @@ async def list_services(
     # 3. then we compose the final service using as a base the registry service, overriding with the same
     #    service from the database, adding also the access rights and the owner as email address instead of gid
     # NOTE: this final step runs in a process pool so that it runs asynchronously and does not block in any way
-    with ProcessPoolExecutor(max_workers=2) as pool:
+    with non_blocking_process_pool_executor(max_workers=2) as pool:
```

Comment:
why not default executor?

Reply:
This was done in via ProcessPool I've maintained the same pattern to avoid issues.

Reply:
I was confused. I thought there was a default process pool as well. Therefore my question.